bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java

```diff
@@ -139,8 +139,10 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {
         int result = super.hashCode();
-        result = 31 * result + Arrays.hashCode(paramTypes);
-        result = 31 * result + retType.hashCode();
+        if (!SymbolFlags.isFlagOn(this.flags, SymbolFlags.ANY_FUNCTION)) {
```

Comment:
We can return early if `SymbolFlags.isFlagOn(this.flags, SymbolFlags.ANY_FUNCTION)` and get rid of the indentation.